### PR TITLE
feat: ensure demo users are in the limited demo group only (ENG-4256)

### PR DESCRIPTION
**Pull request by albertodonato**

### CODE REVIEW

**Review by fwereade — APPROVED** [review]
LGTM – hoping we'll be able to build it sooner rather than later :).

File: redash/authentication/__init__.py [hunk 1]
```diff
@@ -226,6 +230,54 @@ def jwt_token_load_user_from_request(request):
     return user
 
 
+def ensure_demo_group(org):
+    group_name = "demo"
+    group_permissions = [
+        "create_query",
```
Comment (fwereade) [inline redash/authentication/__init__.py]: do we want this one? (I presume it's for testing)

Comment (albertodonato) [reply redash/authentication/__init__.py]: I've removed it, since we don't want demo users to run arbitrary queries

**Review by fwereade — APPROVED** [review]
LGTM, thanks

**Pull request merged**